torch/autograd/gradcheck.py

```diff
@@ -15,17 +15,26 @@ def zero_gradients(x):
         for elem in x:
             zero_gradients(elem)
 
-
-def make_jacobian(input, num_out):
+# Returns a tensor or iterable of tensors each with shape
+# (input.nelement(), num_out) if input is of non-complex dtype
+# and of shape (input.nelement(), num_out, 2) for tensors with complex dtype when extra_dim_if_complex
+# is set to True. The extra dimension helps store both wirtinger and conjugate wirtinger derivative
+# required for gradcheck for complex functions.
+# For R -> C, wirtinger and conjugate wirtinger derivative are equal.
+def make_jacobian(input, num_out, extra_dim_if_complex=False):
```

Comment:
What is the reason for this flag? Are there cases where you don't want to do this?

Reply:
yeah we don't wanna do this when this function is called in `get_analytical_jacobian`